v2/cmd/car/list.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	carv2 "github.com/ipld/go-car/v2"
+	icarv1 "github.com/ipld/go-car/v2/internal/carv1"
+	"github.com/urfave/cli/v2"
+)
+
+// ListCar is a command to output the cids in a car.
+func ListCar(c *cli.Context) error {
+	r, err := carv2.OpenReader(c.Args().Get(0))
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+
+	outStream := os.Stdout
+	if c.Args().Len() >= 2 {
+		outStream, err = os.Create(c.Args().Get(1))
+		if err != nil {
+			return err
+		}
+	}
+	defer outStream.Close()
+
+	rd, err := icarv1.NewCarReader(r.DataReader())
```

Comment:
use https://pkg.go.dev/github.com/ipld/go-car/v2@master#BlockReader? it was written precisely for this :) and it even supports both carv1 and carv2.